src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -461,6 +526,31 @@ internal void ParseSolution()
                 }
             }
 
+            if (_solutionFilter != null)
+            {
+                HashSet<string> projectPaths = new HashSet<string>(_projectsInOrder.Count, NativeMethodsShared.IsLinux ? StringComparer.Ordinal : StringComparer.OrdinalIgnoreCase);
```

Comment:
I don't like sprinkling this case-sensitivity check here, since it's wrong in some cases (you can have a case-sensitive FS on macOS, and you can have case-insensitive ones on Linux). I don't think we need to fix that right now (it's basically impossible to get right), but I'd like to at least have it in one place--should we create a `NativeMethodsShared.IsFileSystemCaseSensitive` or `NativeMethodsShared.FileSystemComparer` or put a comparer into `IFileSystem` and use it here? @cdmihai feel free to chime in :)

Reply:
I replaced it with a call to NativeMethodsShared.OSUsesCaseSensitivePaths instead—for now, that just returns NativeMethodsShared.IsLinux, but if we decide it's time to implement a more complicated but more correct solution, this will be silently fixed.